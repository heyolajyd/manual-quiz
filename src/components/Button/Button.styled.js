import styled from 'styled-components';

export const BaseButton = styled.button`
  background-color: ${(props) => props.theme.colors.tealLight};
  cursor: pointer;
  outline: none;
  border-radius: 24px;
  height: 48px;
  padding: 10px 35px;
  color: ${(props) => props.theme.colors.white};
  font-size: ${props => props.theme.textSizes.small};
  font-family: ${(props) => props.theme.fonts.montserrat};
  font-weight: ${(props) => props.theme.fontWeight.bold};
  letter-spacing: 2.4px;
  border: 0;
  max-width: 220px;
  box-shadow: 0 10px 30px -20px rgba(0, 0, 0, 0.5);

  &:disabled {
    color: rgba(16, 16, 16, 0.3);
    background-color: rgba(239, 239, 239, 0.3);
    border-color: rgba(195, 195, 195, 0.3);
  }
`;
