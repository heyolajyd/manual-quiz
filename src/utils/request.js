/**
 * Util function to get questions
 */
export const getQuestion = async () => {
  try {
    const res = await fetch('./questions.json');
    return await res.json();
  } catch (err) {
    console.log('Failed to parse: ', JSON.stringify(err));
    throw new Error(err);
  }
};
