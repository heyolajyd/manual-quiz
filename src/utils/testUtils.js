import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
