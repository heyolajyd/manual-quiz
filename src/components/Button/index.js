import React from 'react';
import PropTypes from 'prop-types';
import { BaseButton } from './Button.styled';

const Button = (props) => {
  const { onClick, disabled = false, children } = props;

  return (
    <BaseButton onClick={onClick} disabled={disabled}>
      {children}
    </BaseButton>
  );
};

Button.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default Button;
