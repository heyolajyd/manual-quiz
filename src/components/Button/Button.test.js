import React from 'react';
import { fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { withThemeProvider } from 'utils/testUtils';
import Button from './index';

const onClickMock = jest.fn();

describe('Button', function () {
  beforeEach(() => {
    const { container, getByText } = withThemeProvider(
      <Button onClick={onClickMock}>Testing</Button>
    );

    this.container = container;
    this.getByText = getByText;
  });

  it('should match snapshot', () => {
    expect(this.container).toMatchSnapshot();
  });

  it('should be able to click button component', () => {
    fireEvent.click(this.getByText(/Testing/));

    expect(onClickMock).toBeCalled();
  });
});
