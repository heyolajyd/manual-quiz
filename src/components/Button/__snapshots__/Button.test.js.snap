// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
.c0 {
  background-color: #34d1d1;
  cursor: pointer;
  outline: none;
  border-radius: 24px;
  height: 48px;
  padding: 10px 35px;
  color: #fff;
  font-size: 12px;
  font-family: Montserrat;
  font-weight: 700;
  -webkit-letter-spacing: 2.4px;
  -moz-letter-spacing: 2.4px;
  -ms-letter-spacing: 2.4px;
  letter-spacing: 2.4px;
  border: 0;
  max-width: 220px;
  box-shadow: 0 10px 30px -20px rgba(0,0,0,0.5);
}

.c0:disabled {
  color: rgba(16,16,16,0.3);
  background-color: rgba(239,239,239,0.3);
  border-color: rgba(195,195,195,0.3);
}

<div>
  <button
    class="c0"
  >
    Testing
  </button>
</div>
`;
